include/zephyr/bluetooth/conn.h

```diff
@@ -262,6 +262,157 @@ enum __packed bt_conn_type {
+};
+
+/** Supported Sounding Sequence RTT precision. */
+enum bt_conn_le_cs_capability_rtt_sounding {
+	/** Sounding Sequence RTT variant is not supported. */
+	BT_CONN_LE_CS_RTT_SOUNDING_NOT_SUPP = 0,
+	/** 10ns time-of-flight accuracy. */
+	BT_CONN_LE_CS_RTT_SOUNDING_10NS,
+	/** 150ns time-of-flight accuracy. */
+	BT_CONN_LE_CS_RTT_SOUNDING_150NS,
+};
+
+/** Supported Random Payload RTT precision. */
+enum bt_conn_le_cs_capability_rtt_random_payload {
+	/** Random Payload RTT variant is not supported. */
+	BT_CONN_LE_CS_RTT_RANDOM_PAYLOAD_NOT_SUPP = 0,
+	/** 10ns time-of-flight accuracy. */
+	BT_CONN_LE_CS_RTT_RANDOM_PAYLOAD_10NS,
+	/** 150ns time-of-flight accuracy. */
+	BT_CONN_LE_CS_RTT_RANDOM_PAYLOAD_150NS,
+};
+
+/** Remote channel sounding capabilities for LE connections supporting CS */
+struct bt_conn_le_cs_capabilities {
+	/** Number of CS configurations */
+	uint8_t num_config_supported;
+	/** Maximum number of consecutive CS procedures.
+	 *
+	 * When set to zero, indicates support for both fixed and indefinite
+	 * numbers of CS procedures before termination.
+	 */
+	uint16_t max_consecutive_procedures_supported;
+	/** Number of antennas. */
+	uint8_t num_antennas_supported;
+	/** Maximum number of antenna paths. */
+	uint8_t max_antenna_paths_supported;
+	/** Initiator role. */
+	bool initiator_supported;
+	/** Reflector role. */
+	bool reflector_supported;
+	/** Mode-3 */
+	bool mode_3_supported;
+	/** RTT AA-Only */
+	enum bt_conn_le_cs_capability_rtt_aa_only rtt_aa_only_precision;
+	/** RTT Sounding */
+	enum bt_conn_le_cs_capability_rtt_sounding rtt_sounding_precision;
+	/** RTT Random Payload */
+	enum bt_conn_le_cs_capability_rtt_random_payload rtt_random_payload_precision;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT AA Only.
+	 *
+	 * Set to 0 if RTT AA Only isn't supported.
+	 */
+	uint8_t rtt_aa_only_n;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT Sounding.
+	 *
+	 * Set to 0 if RTT Sounding isn't supported
+	 */
+	uint8_t rtt_sounding_n;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT Random Payload.
+	 *
+	 * Set to 0 if RTT Random Payload isn't supported.
+	 */
+	uint8_t rtt_random_payload_n;
+	/** Phase-based normalized attack detector metric
+	 * when a CS_SYNC with sounding sequence is received.
+	 */
+	bool phase_based_nadm_sounding_supported;
+	/** Phase-based normalized attack detector metric
+	 * when a CS_SYNC with random sequence is received.
+	 */
+	bool phase_based_nadm_random_supported;
+	/** CS_SYNC LE 2M PHY. */
+	bool cs_sync_2m_phy_supported;
+	/** CS_SYNC LE 2M 2BT PHY. */
+	bool cs_sync_2m_2bt_phy_supported;
+	/** Subfeature: CS with no Frequency Actuation Error. */
+	bool cs_without_fae_supported;
+	/** Subfeature: Channel Selection Algorithm #3c */
+	bool chsel_alg_3c_supported;
+	/** Subfeature: Phase-based Ranging from RTT sounding sequence. */
+	bool pbr_from_rtt_sounding_seq_supported;
```

Comment:
I'd still like to hear some rationale for going with bools instead of bitfields. I'm not saying bools is wrong, but it certainly increases the struct size substantially (8 bits instead of 1 bit for each feature).

Reply:
I think it's slightly easier to read code that uses booleans compared to bitmasking. No strong preference however.

For the parameters with many valid bits (T_XX) I used bitfields as it does become a bit cumbersome to unpack those. For the parameters with only a couple bits (role support, mode support, NADM, PHYs, Subfeatures) I unpacked them to booleans.

size comparison for the relevant struct elements:
- pure hci format: 1 role + 1 mode + 1 phys + 2 subfeatures + 2 NADM sounding + 2 NADM random = 9 bytes
- with booleans / this PR currently: 2 role + 1 mode + 2 phys + 3 subfeatures + 1 NADM sounding + 1 NADM random = 10 bytes

Reply:
Ok. That's not too bad then.